### DISCUSSION TITLE
Fresh install, doesn't work. Logs included. Please help

### DISCUSSION THREAD

**Discussion by gap579137** [plausible/analytics#807] [Self-Hosted Support] [answered]
root@localhost:~/hosting# docker-compose up
hosting_mail_1 is up-to-date
hosting_plausible_db_1 is up-to-date
hosting_plausible_events_db_1 is up-to-date
Starting hosting_plausible_1 ... done
Attaching to hosting_mail_1, hosting_plausible_db_1, hosting_plausible_events_db_1, hosting_plausible_1
mail_1                 |     1 LOG: MAIN
mail_1                 |     1   exim 4.89 daemon started: pid=1, -q30m, listening for SMTP on port 25 (IPv6 and IPv4)
plausible_db_1         | The files belonging to this database system will be owned by user "postgres".
plausible_db_1         | This user must also own the server process.
plausible_db_1         |
plausible_db_1         | The database cluster will be initialized with locale "en_US.utf8".
plausible_db_1         | The default database encoding has accordingly been set to "UTF8".
plausible_db_1         | The default text search configuration will be set to "english".
plausible_db_1         |
plausible_db_1         | Data page checksums are disabled.
plausible_db_1         |
plausible_db_1         | fixing permissions on existing directory /var/lib/postgresql/data ... ok
plausible_db_1         | creating subdirectories ... ok
plausible_db_1         | selecting dynamic shared memory implementation ... posix
plausible_db_1         | selecting default max_connections ... 100
plausible_db_1         | selecting default shared_buffers ... 128MB
plausible_db_1         | selecting default time zone ... Etc/UTC
plausible_db_1         | creating configuration files ... ok
plausible_db_1         | running bootstrap script ... ok
plausible_db_1         | performing post-bootstrap initialization ... ok
plausible_db_1         | syncing data to disk ... ok
plausible_db_1         |
plausible_db_1         | initdb: warning: enabling "trust" authentication for local connections
plausible_db_1         | You can change this by editing pg_hba.conf or using the option -A, or
plausible_db_1         | --auth-local and --auth-host, the next time you run initdb.
plausible_db_1         |
plausible_db_1         | Success. You can now start the database server using:
plausible_db_1         |
plausible_db_1         |     pg_ctl -D /var/lib/postgresql/data -l logfile start
plausible_db_1         |
plausible_db_1         | waiting for server to start....2021-03-08 14:42:04.050 UTC [46] LOG:  starting PostgreSQL 12.6 (Debian 12.6-1.pgdg100+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit
plausible_db_1         | 2021-03-08 14:42:04.050 UTC [46] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
plausible_db_1         | 2021-03-08 14:42:04.064 UTC [47] LOG:  database system was shut down at 2021-03-08 14:42:03 UTC
plausible_db_1         | 2021-03-08 14:42:04.067 UTC [46] LOG:  database system is ready to accept connections
plausible_db_1         |  done
plausible_db_1         | server started
plausible_db_1         |
plausible_db_1         | /usr/local/bin/docker-entrypoint.sh: ignoring /docker-entrypoint-initdb.d/*
plausible_db_1         |
plausible_db_1         | 2021-03-08 14:42:04.220 UTC [46] LOG:  received fast shutdown request
plausible_db_1         | waiting for server to shut down....2021-03-08 14:42:04.220 UTC [46] LOG:  aborting any active transactions
plausible_db_1         | 2021-03-08 14:42:04.224 UTC [46] LOG:  background worker "logical replication launcher" (PID 53) exited with exit code 1
plausible_db_1         | 2021-03-08 14:42:04.224 UTC [48] LOG:  shutting down
plausible_db_1         | 2021-03-08 14:42:04.230 UTC [46] LOG:  database system is shut down
plausible_db_1         |  done
plausible_db_1         | server stopped
plausible_db_1         |
plausible_db_1         | PostgreSQL init process complete; ready for start up.
plausible_db_1         |
plausible_db_1         | 2021-03-08 14:42:04.332 UTC [1] LOG:  starting PostgreSQL 12.6 (Debian 12.6-1.pgdg100+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit
plausible_db_1         | 2021-03-08 14:42:04.334 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
plausible_db_1         | 2021-03-08 14:42:04.334 UTC [1] LOG:  listening on IPv6 address "::", port 5432
plausible_db_1         | 2021-03-08 14:42:04.335 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
plausible_db_1         | 2021-03-08 14:42:04.346 UTC [65] LOG:  database system was shut down at 2021-03-08 14:42:04 UTC
plausible_db_1         | 2021-03-08 14:42:04.348 UTC [1] LOG:  database system is ready to accept connections
plausible_events_db_1  | Processing configuration file '/etc/clickhouse-server/config.xml'.
plausible_events_db_1  | Merging configuration file '/etc/clickhouse-server/config.d/docker_related_config.xml'.
plausible_events_db_1  | Merging configuration file '/etc/clickhouse-server/config.d/logging.xml'.
plausible_events_db_1  | Logging warning to /var/log/clickhouse-server/clickhouse-server.log
plausible_events_db_1  | Logging errors to /var/log/clickhouse-server/clickhouse-server.err.log
plausible_events_db_1  | Logging warning to console
plausible_events_db_1  | 2021.03.08 14:42:03.519302 [ 45 ] {} <Warning> Access(local directory): File /var/lib/clickhouse/access/users.list doesn't exist
plausible_events_db_1  | 2021.03.08 14:42:03.519587 [ 45 ] {} <Warning> Access(local directory): Recovering lists in directory /var/lib/clickhouse/access/
plausible_events_db_1  | 2021.03.08 14:42:03.566470 [ 45 ] {} <Warning> Application: Listen [::]:8123 failed: Poco::Exception. Code: 1000, e.code() = 0, e.displayText() = DNS error: EAI: Address family for hostname not supported (version 21.2.5.5 (official build)). If it is an IPv6 or IPv4 address and your host has disabled IPv6 or IPv4, then consider to specify not disabled IPv4 or IPv6 address to listen in <listen_host> element of configuration file. Example for disabled IPv6: <listen_host>0.0.0.0</listen_host> . Example for disabled IPv4: <listen_host>::</listen_host>
plausible_events_db_1  | 2021.03.08 14:42:03.566964 [ 45 ] {} <Warning> Application: Listen [::]:9000 failed: Poco::Exception. Code: 1000, e.code() = 0, e.displayText() = DNS error: EAI: Address family for hostname not supported (version 21.2.5.5 (official build)). If it is an IPv6 or IPv4 address and your host has disabled IPv6 or IPv4, then consider to specify not disabled IPv4 or IPv6 address to listen in <listen_host> element of configuration file. Example for disabled IPv6: <listen_host>0.0.0.0</listen_host> . Example for disabled IPv4: <listen_host>::</listen_host>
plausible_events_db_1  | 2021.03.08 14:42:03.567341 [ 45 ] {} <Warning> Application: Listen [::]:9009 failed: Poco::Exception. Code: 1000, e.code() = 0, e.displayText() = DNS error: EAI: Address family for hostname not supported (version 21.2.5.5 (official build)). If it is an IPv6 or IPv4 address and your host has disabled IPv6 or IPv4, then consider to specify not disabled IPv4 or IPv6 address to listen in <listen_host> element of configuration file. Example for disabled IPv6: <listen_host>0.0.0.0</listen_host> . Example for disabled IPv4: <listen_host>::</listen_host>
plausible_events_db_1  | 2021.03.08 14:42:03.567702 [ 45 ] {} <Warning> Application: Listen [::]:9004 failed: Poco::Exception. Code: 1000, e.code() = 0, e.displayText() = DNS error: EAI: Address family for hostname not supported (version 21.2.5.5 (official build)). If it is an IPv6 or IPv4 address and your host has disabled IPv6 or IPv4, then consider to specify not disabled IPv4 or IPv6 address to listen in <listen_host> element of configuration file. Example for disabled IPv6: <listen_host>0.0.0.0</listen_host> . Example for disabled IPv4: <listen_host>::</listen_host>
plausible_1            | ERROR! Config provider Config.Reader failed with:
plausible_1            | ** (RuntimeError) BASE_URL must start with `http` or `https`. Currently configured as `replace-me`
plausible_1            |     (stdlib 3.14) erl_eval.erl:680: :erl_eval.do_apply/6
plausible_1            |     (elixir 1.11.3) src/elixir.erl:280: :elixir.recur_eval/3
plausible_1            |     (elixir 1.11.3) src/elixir.erl:265: :elixir.eval_forms/3
plausible_1            |     (elixir 1.11.3) lib/code.ex:341: Code.eval_string_with_error_handling/3
plausible_1            |     (elixir 1.11.3) lib/config.ex:255: Config.__eval__!/3
plausible_1            |     (elixir 1.11.3) lib/config/reader.ex:86: Config.Reader.read!/2
plausible_1            |     (elixir 1.11.3) lib/config/reader.ex:52: Config.Reader.load/2
plausible_1            |     (elixir 1.11.3) lib/config/provider.ex:347: anonymous fn/2 in Config.Provider.run_providers/2
plausible_1            |
plausible_1            | {"init terminating in do_boot",{#{'__exception__'=>true,'__struct__'=>'Elixir.RuntimeError',message=><<"BASE_URL must start with `http` or `https`. Currently configured as `replace-me`">>},[{erl_eval,do_apply,6,[{file,"erl_eval.erl"},{line,680}]},{elixir,recur_eval,3,[{file,"src/elixir.erl"},{line,280}]},{elixir,eval_forms,3,[{file,"src/elixir.erl"},{line,265}]},{'Elixir.Code',eval_string_with_error_handling,3,[{file,"lib/code.ex"},{line,341}]},{'Elixir.Config','__eval__!',3,[{file,"lib/config.ex"},{line,255}]},{'Elixir.Config.Reader','read!',2,[{file,"lib/config/reader.ex"},{line,86}]},{'Elixir.Config.Reader',load,2,[{file,"lib/config/reader.ex"},{line,52}]},{'Elixir.Config.Provider','-run_providers/2-fun-0-',2,[{file,"lib/config/provider.ex"},{line,347}]}]}}
plausible_1            | init terminating in do_boot ({,[{erl_eval,do_apply,6,[{_},{_}]},{elixir,recur_eval,3,[{_},{_}]},{elixir,eval_forms,3,[{_},{_}]},{Elixir.Code,eval_string_with_error_handling,3,[{_},{_}]},{Elixir.Config
plausible_1            |
plausible_1            | Crash dump is being written to: erl_crash.dump...done
hosting_plausible_1 exited with code 1

**Comment by Philipinho** [comment, marked as answer]
From your logs, I can see 
	`plausible_1 | ** (RuntimeError) *BASE_URL must start with http or https*. Currently configured as replace-me`

Fix and try again.

**Reply by gap579137** [reply]
I was able to get that part fixed and can now access the backend. Thank you! Unfortunatly I am unable to get the tracking script to work. When I follow the link to the to the JS it is unable to connect and the front end does not work.

**Reply by Philipinho** [reply]
By "it does not work", do you mean the url does not load?

**Reply by gap579137** [reply]
I am getting "refused to connect." when I use the IP as the base URL and when I use the domain I have as the base URL I still get "refused to connect."

**Reply by Philipinho** [reply]
Did you add port 8000 to the base_url?

**Reply by gap579137** [reply]
When trying to access it? yes

**Reply by Philipinho** [reply]
`BASE_URL=http://127.0.0.1:8000`

Then re-run` docker-compose up --detach`

**Reply by gap579137** [reply]
I was able to get the back end working again. The site is still not gathering info. I appreciate the help but will likely move forward with a different solution.

**Reply by Philipinho** [reply]
You need to serve the tracking url with https because browsers will block the http link due to "Mixed content".
You should set up a reverse proxy on top of Plausible and then use certbot to generate SSL.

You are actually closer to the finish line.

I will be here to guide you till the end, hopefully.
Feel free to ask me any more questions.

**Reply by gap579137** [reply]
I hate to use up anymore of your time.

**Reply by Philipinho** [reply]
Are you using a custom domain/subdomain already?

**Reply by gap579137** [reply]
subdomain pointed with an "A" record

**Reply by Philipinho** [reply]
Great. Apache or Nginx?

**Reply by gap579137** [reply]
Neither right now. It was a fresh install of Ubuntu 20.04

**Reply by Philipinho** [reply]
How do you then access your site via the internet?

**Reply by Philipinho** [reply]
Run `systemctl status apache2` and `systemctl status nginx` to confirm if any of these web servers were pre-installed.

**Reply by gap579137** [reply]
currently have an "A" record setup up with the namecheap that is set to point to the IP address of the Linode VPS

**Reply by gap579137** [reply]
Neither are installed

**Reply by Philipinho** [reply]
Here is what to do.
The below code will install and configure Nginx reverse proxy, Certbot for Letsencrypt SSL and firewall.

Create a file `install.sh` . Copy the below code into the file.
replace `your.domain.com` with your own domain.
replace `email@domain.com` with your email and then save the file.

```
custom_domain=your.domain.com
certbot_email=email@domain.com

sudo apt -qqy  update
sudo apt -qqy install nginx
sudo apt -qqy install certbot python3-certbot-nginx ufw
sudo systemctl enable nginx
sudo systemctl start nginx

ufw allow ssh && ufw allow http && ufw allow https && ufw allow 8000 && ufw --force enable

sudo wget https://raw.githubusercontent.com/plausible/hosting/master/reverse-proxy/nginx/plausible -O /etc/nginx/sites-available/plausible
sudo sed -i "s/example.com/$custom_domain/" /etc/nginx/sites-available/plausible

sudo ln -s /etc/nginx/sites-available/plausible /etc/nginx/sites-enabled/plausible
	
sudo systemctl reload nginx

sudo certbot --nginx -d $custom_domain --email $certbot_email --agree-tos --no-eff-email
echo "Installation complete"
```
Run `bash install.sh` and wait for the script to run.

Next, you will be prompted by cert bot to make all requests redirect to secure HTTPS access.
Select option` 2` and click enter.

When you are done with that. You should be able to access your site via https.

Next, change Plausible's  BASE_URL to `BASE_URL=https://your.domain.com`  (without the port).
Then re-run `docker-compose up --detach`

This should pretty much get the job done.

Do not hesitate to reach out if you encounter any issues.

**Reply by gap579137** [reply]
This worked perfectly! Thank you
